onnxruntime/core/providers/webgpu/buffer_manager.cc

```diff
@@ -256,10 +256,10 @@ void BufferManager::Upload(void* src, WGPUBuffer dst, size_t size) {
   memcpy(mapped_data, src, size);
   staging_buffer.Unmap();
 
-  auto& command_encoder = context_.GetCommandEncoder();
-  context_.EndComputePass();
+  auto command_encoder = context_.Device().CreateCommandEncoder();
   command_encoder.CopyBufferToBuffer(staging_buffer, 0, dst, 0, buffer_size);
-  pending_staging_buffers_.push_back(staging_buffer);
+  auto command_buffer = command_encoder.Finish();
+  context_.Device().GetQueue().Submit(1, &command_buffer);
```

Comment:
why not just free the staging_buffer here?

Reply:
`staging_buffer` is of type `wgpu::Buffer` and it's expected to be freed in the destructor ( `wgpuBufferRelease` is called )